dashboard/src/actions/stripe.ts

```diff
@@ -0,0 +1,55 @@
+'use server';
+
+import { withUserAuth } from '@/auth/auth-actions';
+import { stripe } from '@/lib/billing/stripe';
+import { SelectedPlan, SelectedPlanSchema } from '@/types/pricing';
+import { getAuthSession } from '@/auth/auth-actions';
+import { getLookupKeyFromTierConfig } from '@/lib/billing/plans';
+
+export const createStripeCheckoutSession = withUserAuth(async (userId: string, planData: SelectedPlan) => {
+  try {
+    const validatedPlan = SelectedPlanSchema.parse(planData);
+
+    if (validatedPlan.price_cents === 0 && validatedPlan.tier === 'growth') {
+      throw new Error('Free plans do not require checkout');
+    }
+
+    if (validatedPlan.tier === 'enterprise') {
+      throw new Error('Custom plans require manual setup');
+    }
+
```

Comment:
Suggestion/Question: I don't know what's the "best practice" regarding this, so I'd love to hear your input on it.
Instead of doing the "manual" validation of the `SelectPlanSchema`, after "validating" it:
Would it make sense to do the following?
1. Add the `tier === 'enterprice'` validation on the `SelectPlanSchema` schema
2. Create a `SelectInitialPlanSchema` that extends `SelectPlanSchema` with the pricing / growth validation

Or is it better to separate the business logic from the schema validation, like it is now? 😄